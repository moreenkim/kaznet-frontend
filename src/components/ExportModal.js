import React, { Component } from "react";
import ExportForm from "./ExportForm";
import { Modal, ModalHeader, ModalBody } from "reactstrap";

export default class ExportModal extends Component {
  render() {
    return (
      <div>
        <Modal
          isOpen={this.props.modalState}
          toggle={this.props.downloadModalHandler}
          className={this.props.className}
        >
          <ModalHeader toggle={this.props.downloadModalHandler}>
            Submissions of {this.props.userName}
          </ModalHeader>
          <ModalBody>
            <p className="font-weight-normal">Export Submissions </p>
            <ExportForm
              onFormSubmit={this.props.onFormSubmit}
              downloadModalHandler={this.props.downloadModalHandler}
            />
          </ModalBody>
        </Modal>
      </div>
    );
  }
}
