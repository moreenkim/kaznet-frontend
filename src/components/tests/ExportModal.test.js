// test ExportModal
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MockDate from 'mockdate';
import ExportModal from '../ExportModal';

describe('components/ExportModal', () => {
  it('renders without crashing', () => {
    shallow(
      <ExportModal
        isOpen={function() {}}
        toggle={function() {}}
        className="some-fancy-class"
        onFormSubmit={function() {}}
        downloadModalHandler={function() {}}
      />
    );
  });
  it('renders an ExportModal', () => {
    MockDate.set('1/2/1986');
    const wrapper = mount(
      <ExportModal
        isOpen={function() {}}
        toggle={function() {}}
        className="some-fancy-class"
        onFormSubmit={function() {}}
        downloadModalHandler={function() {}}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
