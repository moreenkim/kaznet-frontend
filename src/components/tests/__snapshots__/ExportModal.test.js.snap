// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ExportModal renders an ExportModal 1`] = `
<ExportModal
  className={[Function]}
  downloadModalHandler={[Function]}
  isOpen={[Function]}
  onFormSubmit={[Function]}
  toggle={[Function]}
>
  <div>
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      centered={false}
      className={[Function]}
      fade={true}
      isOpen={false}
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      role="dialog"
      toggle={[Function]}
      zIndex={1050}
    />
  </div>
</ExportModal>
`;
